--- privoxy-3.0.10-stable/GNUmakefile.in~	2008-10-14 19:47:45.000000000 +0200
+++ privoxy-3.0.10-stable/GNUmakefile.in	2008-10-14 19:48:00.000000000 +0200
@@ -1139,7 +1139,7 @@
 			$(RM) $(DESTDIR)$(CONF_DEST)/$$i ;\
 			$(ECHO) Installing fresh $$i;\
 			$(INSTALL) $$INSTALL_CONF $$i $(DESTDIR)$(CONF_DEST) || exit 1;\
-		elif [ -s "$(CONF_DEST)/$$i" ]; then \
+		elif [ -s "$(DESTDIR)$(CONF_DEST)/$$i" ]; then \
 			$(ECHO) Installing $$i as $$i.new ;\
 			$(INSTALL) $$INSTALL_CONF $$i $(DESTDIR)$(CONF_DEST)/$$i.new || exit 1;\
 			NEW=1;\
